File: Python/3084.py
import sys

for line in sys.stdin:
    try:
        entrada = line.split()
        entrada = [int(valor) for valor in entrada] 
        hora = entrada[0]//30
        minuto = entrada[1]//6
    
        if hora<10:
            if minuto<10:
                print(f"0{hora}:0{minuto}")
            else:
                print(f"0{hora}:{minuto}")
        
        else:
            if minuto<10:
                print(f"{hora}:0{minuto}")
            else:
                print(f"{hora}:{minuto}")
    except EOFError:
        sys.exit()
    
    

File: C/3084.c
#include <stdio.h>

int main()
{
    int x, d, hora, minuto;

    while(scanf("%d%d", &x, &d)!=EOF){
        hora=x/30;
        minuto=d/6;
        
        if(hora<10){
            if(minuto<10){
                printf("0%d:0%d\n", hora, minuto);
            }else{
                printf("0%d:%d\n", hora, minuto);
            }
        }else{
            if(minuto<10){
                printf("%d:0%d\n", hora, minuto);
            }else{
                printf("%d:%d\n", hora, minuto);
            }
        }
    }
    return 0;
}
